File: notification-schema-java/src/main/java/uk/gov/defra/tracesx/notificationschema/representation/ApprovedEstablishment.java
package uk.gov.defra.tracesx.notificationschema.representation;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import uk.gov.defra.tracesx.notificationschema.validation.ErrorCodes;
import uk.gov.defra.tracesx.notificationschema.validation.groups.NotificationCvedaFieldValidation;
import uk.gov.defra.tracesx.notificationschema.validation.groups.NotificationHighRiskFieldValidation;

import java.util.List;
import javax.validation.constraints.NotNull;

@Builder
@Data
@JsonInclude(Include.NON_EMPTY)
@NoArgsConstructor(access = AccessLevel.PUBLIC)
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class ApprovedEstablishment {

  private String id;

  @NotNull(groups = NotificationCvedaFieldValidation.class, message = ErrorCodes.NOT_NULL)
  private String name;

  @NotNull(groups = NotificationHighRiskFieldValidation.class, message = ErrorCodes.NOT_NULL)
  private String country;

  private List<String> types;
  private String approvalNumber;
  private String section;
}

File: imports-frontend-entities/src/entities/approved_establishment.js
const _ = require('lodash')
const handler = require('./base/handler')

module.exports = class ApprovedEstablishment {
  constructor(obj) {

    if (!obj) {
      obj = {}
    }
    this.name = obj.name
    this.country = obj.country
    this.approvalNumber = obj.approvalNumber
    this.types = _.get(obj, 'types', [])
    this.section = obj.section
    return Object.seal(new Proxy(this, handler))
  }
}